File: app/src/main/java/com/vegasoft/mypasswords/presentation/LoginActivity.java
package com.vegasoft.mypasswords.presentation;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.KeyEvent;
import android.view.View;
import android.view.View.OnClickListener;
import android.view.inputmethod.EditorInfo;
import android.widget.EditText;
import android.widget.TextView;

import com.vegasoft.mypasswords.R;
import com.vegasoft.mypasswords.bussiness.ConfigManager;

public class LoginActivity extends AppCompatActivity {

    private EditText mPasswordView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);
        // Set up the login form.

        mPasswordView = findViewById(R.id.password);
        mPasswordView.setOnEditorActionListener(new TextView.OnEditorActionListener() {
            @Override
            public boolean onEditorAction(TextView textView, int id, KeyEvent keyEvent) {
                if (id == EditorInfo.IME_ACTION_DONE || id == EditorInfo.IME_NULL) {
                    attemptLogin();
                    return true;
                }
                return false;
            }
        });

        findViewById(R.id.email_sign_in_button).setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View view) {
                attemptLogin();
            }
        });
    }

    private void attemptLogin() {
        mPasswordView.setError(null);

        String password = mPasswordView.getText().toString();
        if (!isPasswordValid(password)) {
            mPasswordView.setError(getString(R.string.error_invalid_password));
            return;
        }
        startActivity(new Intent(this, MainActivity.class));
        finish();
    }

    private boolean isPasswordValid(String password) {
        return password.length() > 4 && new ConfigManager(this).getPSWD().equals(password);
    }
}

File: app/src/main/java/com/vegasoft/mypasswords/presentation/LoginActivity.kt
package com.vegasoft.mypasswords.presentation

import android.content.Intent
import android.os.Bundle
import android.text.TextUtils
import android.view.inputmethod.EditorInfo
import android.widget.TextView.OnEditorActionListener
import androidx.appcompat.app.AppCompatActivity
import com.vegasoft.mypasswords.R
import com.vegasoft.mypasswords.bussiness.ConfigManager
import kotlinx.android.synthetic.main.activity_login.password as mPasswordView
import kotlinx.android.synthetic.main.activity_login.email_sign_in_button

class LoginActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)
        mPasswordView.setOnEditorActionListener(OnEditorActionListener { _, id, _ ->
            if (id == EditorInfo.IME_ACTION_DONE || id == EditorInfo.IME_NULL) {
                attemptLogin()
                return@OnEditorActionListener true
            }
            false
        })
        actionBar?.hide()
        email_sign_in_button.setOnClickListener { attemptLogin() }
        if (isPswdSet()) {
            startActivity(Intent(this, MainActivity::class.java))
            finish()
            return
        }
    }

    private fun attemptLogin() {
        mPasswordView.error = null
        val password = mPasswordView.text.toString()
        if (!isPasswordValid(password)) {
            mPasswordView.error = getString(R.string.error_invalid_password)
            return
        }

        if (!isPasswordCorrect(password)) {
            mPasswordView.error = getString(R.string.error_incorrect_password)
            return
        }
        startActivity(Intent(this, MainActivity::class.java))
        finish()
    }

    private fun isPswdSet() = TextUtils.isEmpty(ConfigManager(this).pswd)

    private fun isPasswordCorrect(password: String) = ConfigManager(this).pswd == password

    private fun isPasswordValid(password: String) = password.length >= 4
}